packages/grid/x-grid/src/tests/rows.XGrid.test.tsx

```diff
@@ -468,4 +472,74 @@ describe('<XGrid /> - Rows', () => {
       });
     });
   });
+
+  describe('Cell focus', () => {
+    let apiRef: GridApiRef;
+
+    const TestCase: React.FC<Pick<XGridProps, 'rows'>> = ({ rows }) => {
+      apiRef = useGridApiRef();
+
+      return (
+        <div style={{ width: 300, height: 300 }}>
+          <XGrid apiRef={apiRef} {...baselineProps} rows={rows} />
+        </div>
+      );
+    };
+
+    beforeEach(() => {
+      baselineProps = {
+        autoHeight: isJSDOM,
+        rows: [
+          {
+            id: 1,
+            clientId: 'c1',
+            first: 'Mike',
+            age: 11,
+          },
+          {
+            id: 2,
+            clientId: 'c2',
+            first: 'Jack',
+            age: 11,
+          },
+          {
+            id: 3,
+            clientId: 'c3',
+            first: 'Mike',
+            age: 20,
+          },
+        ],
+        columns: [{ field: 'clientId' }, { field: 'first' }, { field: 'age' }],
+      };
+    });
+
+    it('should focus the clicked cell in the state', () => {
+      render(<TestCase rows={baselineProps.rows} />);
+
+      fireEvent.click(getCell(0, 0));
+      expect(apiRef.current.getState().focus.cell).to.deep.equal({
+        id: baselineProps.rows[0].id,
+        field: baselineProps.columns[0].field,
+      });
+    });
+
+    it('should reset focus when removing the row containing the focus cell', async () => {
```

Comment:
```suggestion
    it('should reset focus when removing the row containing the focus cell', () => {
```

Reply:
1. I think that there is a broader subject than just row removal here indeed. Could there be the same behavior for other things than just focus ?

2. I also think so.